### DISCUSSION TITLE
Raptorescort mission completes if the mudraptor (Rex) dies

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14179] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

If Rex dies during the Raptor Escort mission but not the owner, it is still valid.
The event `manandhisraptor2` needs a check at the end to make sure Rex is alive

At the end of the event, add this:

```xml
        <CheckConditionalAction targettag="rex" logicaloperator="And">
          <Conditional IsDead="False" />
          <Conditional InPlayerSubmarine="true" />
          <Failure>
            <SetDataAction identifier="Rexdied" operation="set" value="1" />
          </Failure>
        </CheckConditionalAction>
```

In raptorescort  mission, add this

```xml
      <CompleteCheckDataAction identifier="Rexdied" condition="neq 1"/>
```

This should work? I've heard there's a desync in MP so test this as well
      

### Reproduction steps

Kill Rex

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.5.8.0 (Summer Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by A2ur3** [comment]
mission status desync should be fixed in Summer Update👍

> - Fixed clients sometimes displaying missions as having failed even though they succeeded server-side.

**Reply by Regalis11** [reply]
I don't think that addresses the issue unfortunately: this isn't a problem with desync, the death of the mudraptor doesn't count server-side or in single player either.

EDIT: ah, you were referring to the note about the suggested fix, nvm :D

**Comment by Regalis11** [comment]
Fixed in https://github.com/FakeFishGames/Barotrauma-development/commit/5d587121e792194e487053b6b0663bd9ee9287de (commit in our private repo).

Just a note: you can also use true/false with `SetDataAction`: in this case there's no need to use an integer when the value is always either true or false.
